kong/cmd/utils/prefix_handler.lua

```diff
@@ -388,16 +400,16 @@ local function write_process_secrets_file(path, data)
   return true
 end
 
-local function compile_kong_conf(kong_config)
-  return compile_conf(kong_config, kong_nginx_template)
+local function compile_kong_conf(kong_config, template_env_inject)
+  return compile_conf(kong_config, kong_nginx_template, template_env_inject)
 end
 
 local function compile_kong_gui_include_conf(kong_config)
   return compile_conf(kong_config, kong_nginx_gui_include_template)
 end
 
-local function compile_kong_stream_conf(kong_config)
-  return compile_conf(kong_config, kong_nginx_stream_template)
+local function compile_kong_stream_conf(kong_config, template_env_inject)
```

Comment:
Just an idea: we could make this parameter a part of the template environment and let the template decide what to do with the passing-in variables: https://github.com/Kong/kong-ee/blob/6b28fdc4e188ee2da2d0b64fe032f50a19d6acb6/kong/cmd/utils/prefix_handler.lua#L313

Reply:
isn't that what we are already doing [here](https://github.com/Kong/kong/blob/505902c1753ef323ea41a5b8c42679af233406e9/kong/cmd/utils/prefix_handler.lua#L254)? Or am I misinterpreting?

Reply:
@samugi Ah yes I overlooked the change.